Neos.Flow/Classes/Mvc/ActionResponse.php

```diff
@@ -103,6 +109,32 @@ public function setStatusCode(int $statusCode): void
         $this->statusCode = $statusCode;
     }
 
+    /**
+     * Set a cookie in the HTTP response
+     * This leads to a corresponding `Set-Cookie` header to be set in the HTTP response
+     *
+     * @param Cookie $cookie Cookie to be set in the HTTP response
+     * @api
+     */
+    public function setCookie(Cookie $cookie): void
+    {
+        $this->cookies[$cookie->getName()] = $cookie;
```

Comment:
Should be maybe clone the $cookie object here, in order to prevent changing the instance after the `setCookie` call, possibly even after it is merged and serialized into the Http Response?

Reply:
```suggestion
        $this->cookies[$cookie->getName()] = clone $cookie;
```